src/lib.rs

```diff
@@ -945,24 +946,31 @@ fn fill_todo(
             let dirs = fs::read_dir(path).and_then(|d| {
                 d.map(|e| {
                     e.map(|e| {
-                        let path = if curdir {
-                            PathBuf::from(e.path().file_name().unwrap())
+                        // We sort by filename a few lines below. However, it is actually quite
+                        // expensive to extract the filename from PathBuf. Since we already know the
+                        // filename from `DirEntry` here, we store it proactively (we still
+                        // have to heap allocate it).
```

Comment:
"still"? afaict this adds an allocation. So it's surprising that allocating an extra OsString is still cheaper than getting the filename from the path later.

If you keep the whole `DirEntry` you could call [file_name_ref](https://doc.rust-lang.org/nightly/std/os/unix/fs/trait.DirEntryExt2.html#tymethod.file_name_ref) on `cfg(unix)` to avoid that allocation too.

Reply:
> file_name_ref is unstable, as far as I can see?

Right, too bad.

Regarding the overhead, maybe wait until my rust PR lands and then benchmark again? The `file_name` optimization looks like it just might give us that 20% speedup too.

Reply:
Fair enough, once it gets into nightly, I'll benchmark it against a previous nightly to see if there's a big difference.